huma.go

```diff
@@ -370,11 +370,11 @@ func (r *findResult[T]) EveryPB(pb *PathBuffer, v reflect.Value, f func(reflect.
 
 func findInType[T comparable](t reflect.Type, onType func(reflect.Type, []int) T, onField func(reflect.StructField, []int) T, recurseFields bool, ignore ...string) *findResult[T] {
 	result := &findResult[T]{}
-	_findInType(t, []int{}, result, onType, onField, recurseFields, ignore...)
+	_findInType(t, []int{}, result, onType, onField, recurseFields, make(map[reflect.Type]bool), ignore...)
 	return result
 }
 
-func _findInType[T comparable](t reflect.Type, path []int, result *findResult[T], onType func(reflect.Type, []int) T, onField func(reflect.StructField, []int) T, recurseFields bool, ignore ...string) {
+func _findInType[T comparable](t reflect.Type, path []int, result *findResult[T], onType func(reflect.Type, []int) T, onField func(reflect.StructField, []int) T, recurseFields bool, visited map[reflect.Type]bool, ignore ...string) {
```

Comment:
Does the visited map need to be a boolean? By the looks of it, it only matters if a key exists and not the boolean value, but correct me if I'm wrong! In the case that I'm not, what about:

```suggestion
func _findInType[T comparable](t reflect.Type, path []int, result *findResult[T], onType func(reflect.Type, []int) T, onField func(reflect.StructField, []int) T, recurseFields bool, visited map[reflect.Type]struct{}, ignore ...string) {
```

Reply:
Boolean are cleaner syntactically, plus if you benchmark it has 0 impact on performance.